BOARD=native has incomplete ztimer implementation?  Or missing requirements maybe

#### Description

can not build examples/gcoap-rust for BOARD=native (or native64)

#### Steps to reproduce the issue

git clone 
git checkout -B july2024 2024.07
<install c2rust, llvm, etc.>
cd examples/gcoap-rust
make BOARD=native all

#### Expected results

an executable in bin/

#### Actual results

/bin/ld: /corp/projects/pandora/riotos/RIOT/examples/rust-gcoap/bin/native/periph/timer.o: in function `timer_start':
/corp/projects/pandora/riotos/RIOT/cpu/native/periph/timer.c:205: undefined reference to `timer_settime'
/bin/ld: /corp/projects/pandora/riotos/RIOT/examples/rust-gcoap/bin/native/periph/timer.o: in function `timer_init':
/corp/projects/pandora/riotos/RIOT/cpu/native/periph/timer.c:117: undefined reference to `timer_create'
/bin/ld: /corp/projects/pandora/riotos/RIOT/cpu/native/periph/timer.c:124: undefined reference to `timer_delete'
collect2: error: ld returned 1 exit status
make: *** [/corp/projects/pandora/riotos/RIOT/examples/rust-gcoap/../../Makefile.include:737: /corp/projects/pandora/riotos/RIOT/examples/rust-gcoap/bin/native/rust_gcoap.elf] Error 1
[1]    880012 exit 2     make all BOARD=native

as far as I can see, time_settime() is not a thing anymore?  but native still tries to use it.

#### Versions

Operating system: Linux Ubuntu 20.04
Build environment: 

make print-versions

Operating System Environment
----------------------------
         Operating System: "Ubuntu" "20.04.6 LTS (Focal Fossa)"
                   Kernel: Linux 5.15.0-116-generic x86_64 x86_64
             System shell: /usr/bin/dash (probably dash)
             make's shell: /usr/bin/dash (probably dash)

Installed compiler toolchains
-----------------------------
               native gcc: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
        arm-none-eabi-gcc: missing
                  avr-gcc: missing
           msp430-elf-gcc: missing
       riscv-none-elf-gcc: missing
  riscv64-unknown-elf-gcc: missing
      riscv32-esp-elf-gcc: missing
     xtensa-esp32-elf-gcc: xtensa-esp32-elf-gcc (crosstool-NG esp-2021r2-patch3) 8.4.0
   xtensa-esp32s2-elf-gcc: missing
   xtensa-esp32s3-elf-gcc: missing
   xtensa-esp8266-elf-gcc: missing
                    clang: clang version 10.0.0-4ubuntu1 

Installed compiler libs
-----------------------
     arm-none-eabi-newlib: missing
        msp430-elf-newlib: missing
    riscv-none-elf-newlib: missing
riscv64-unknown-elf-newlib: missing
   riscv32-esp-elf-newlib: missing
  xtensa-esp32-elf-newlib: "3.3.0"
xtensa-esp32s2-elf-newlib: missing
xtensa-esp32s3-elf-newlib: missing
xtensa-esp8266-elf-newlib: missing
                 avr-libc: missing (missing)

Installed development tools
---------------------------
                   ccache: missing
                    cmake: cmake version 3.16.3
                 cppcheck: missing
                  doxygen: 1.8.17
                      git: git version 2.25.1
                     make: GNU Make 4.2.1
                  openocd: missing
                   python: Python 3.8.10
                  python2: Python 2.7.18
                  python3: Python 3.8.10
                   flake8: 6.0.0 (mccabe: 0.7.0, pycodestyle: 2.10.0, pyflakes: 3.0.1) CPython 3.8.10 on
               coccinelle: missing


<!-- Thanks for contributing! -->

--- a/cpu/native/periph/timer.c
+++ b/cpu/native/periph/timer.c
@@ -16,6 +16,8 @@
  * @brief       Native CPU periph/timer.h implementation
  *
  * Uses POSIX realtime clock and POSIX itimer to mimic hardware.
+ * This is done with the timer_settime(3), timer_create(3)  interfaces, which are
+ * sometimes found only in the -lrt library, and not in glibc.
  *
  * This is based on native's hwtimer implementation by Ludwig Knüpfer.
  * I removed the multiplexing, as ztimer does the same. (kaspar)

